### DISCUSSION TITLE
multiple GPUs for training

### DISCUSSION THREAD

**Discussion by saumya221** [ultralytics/yolov5#6625] [Q&A]
I have a quadro P5000 16 GB graphics card in my work station which can support another graphics card as per the workstation manufacturer. My question is do I have to use same graphics card (Quadro P5000 16GB) or similar card having same compute capability of 6.1?
Or can I use another graphics card with different compute capability (For example Quadro RTX 5000 having compute capability of 7.5).

**Comment by glenn-jocher** [comment]
@saumya221 Multi-GPU systems should have identical CUDA devices

**Reply by saumya221** [reply]
@glenn-jocher  One clarification please. When you say identical CUDA devices, do you mean same make and model of GPU or same compute capability GPU?

**Reply by glenn-jocher** [reply]
@saumya221 same make and model. Ideally you should source all your GPU hardware at the same time to avoid any potential issues.

That's not to say you might be able to get different devices to work together, but when something breaks the last thing you want to wonder is if it's because your hardware. You want to reduce risk as much as possible, which on the hardware side means commonality between your CUDA devices.

**Reply by glenn-jocher** [reply]
@saumya221 to give you an example, I know of no cloud resources that you can spinup that offer different GPU types attached to the same instance, i.e. see https://cloud.google.com/compute/docs/gpus
